addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -199,6 +199,38 @@ def _generate_timesheeets(self):
         if results_with_leave_timesheet:
             results_with_leave_timesheet._timesheet_create_lines()
 
+    def _generate_public_time_off_timesheets(self, employees):
+        timesheet_vals_list = []
+        work_hours_data = self._work_time_per_day()
+        timesheet_read_group = self.env['account.analytic.line']._read_group(
+            [('global_leave_id', 'in', self.ids), ('employee_id', 'in', employees.ids)],
+            ['date:array_agg'],
+            ['employee_id']
+        )
+        timesheet_dates_per_employee_id = {
+            res['employee_id'][0]: res['date']
+            for res in timesheet_read_group
+        }
```

Comment:
The result also changed so I think you will have to also change the using of the result.


Probably something like this:
```suggestion
        timesheet_dates_per_employee_id = {
            employee.id: date
            for employee, date in timesheet_read_group
        }
```